pkg/serverless/registration/extension_api.go

```diff
@@ -0,0 +1,78 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package registration
+
+import (
+	"bytes"
+	"fmt"
+	"net/http"
+	"time"
+)
+
+const (
+	extensionName = "datadog-agent"
+	headerExtName = "Lambda-Extension-Name"
+
+	//HeaderExtID is the header name for the extension identifier
+	HeaderExtID = "Lambda-Extension-Identifier"
+)
+
+// RegisterExtension registers the serverless daemon and subscribe to INVOKE and SHUTDOWN messages.
+// Returns either (the serverless ID assigned by the serverless daemon + the api key as read from
+// the environment) or an error.
+func RegisterExtension(runtimeURL string, registrationRoute string, timeout time.Duration) (ID, error) {
```

Comment:
Nice refactor. Should we have the NextInvocation parsing logic in here as well?

Reply:
+1 maybe in an other PR since this one is quite huge already